piet-direct2d/src/dwrite.rs

```diff
@@ -413,6 +425,13 @@ impl TextLayout {
     }
 
     pub fn set_max_width(&mut self, max_width: f64) -> Result<(), Error> {
+        // infinity produces nonsense values for the inking rect on d2d
+        let max_width = if !max_width.is_finite() {
+            MAX_LAYOUT_CONSTRAINT
```

Comment:
What happens if we just don't set a max width? In any case, I think this should be fine.

Reply:
The absence of a max width is represented by `f64::INFINITY` at this point.